README.md

```diff
@@ -933,6 +933,149 @@ _syscall.number()_ returns the system call number referenced by the given `name`
 
 The `syscall.table` library provides support for translating system call names to addresses (light userdata).
 
+### netfilter
+
+The `netfilter` library provides support for the kernel [Netfilter](https://netfilter.org/) subsystem. This includes support for xtables (`match`, `target`) and netfilter hooks.
+
+#### `netfilter.xtable(match, target)`
+
+_netfilter.xtable(ops)_ registers a xtable with the specified match and target operations. The `match` and `target` are tables with the following fields:
+
+* `name`: the name of the xtable.
+* `revision`: the revision of the xtable.
+* `match` or `target`: the callback function that is called when the xtable is matched or targeted. The match callback should return either `NF_DROP` or `NF_ACCEPT` and the target callback should return one of the following values:
+  * `NF_ACCEPT`: accept the packet.
+  * `NF_DROP`: drop the packet.
+  * `NF_STOLEN`: the packet is stolen by the hook.
+  * `NF_QUEUE`: the packet is queued.
+  * `NF_REPEAT`: the packet is repeated.
+  * `NF_STOP`: stop the traversal.
+* `checkentry`: the callback function that is called when the xtable is checked.
+* `destroy`: the callback function that is called when the xtable is destroyed.
+* `family`: the protocol family. This is one of :
+  * `NFPROTO_UNSPEC` (unspecified).
+  * `NFPROTO_INET` (IPv4).
+  * `NFPROTO_IPV4` (IPv4).
+  * `NFPROTO_ARP` (ARP).
+  * `NFPROTO_NETDEV` (network device).
+  * `NFPROTO_BRIDGE` (bridge).
+  * `NFPROTO_IPV6` (IPv6).
+* `proto`: the protocol. This is one of :
+  * `IPPROTO_TCP` (TCP).
+  * `IPPROTO_UDP` (UDP).
+  * `IPPROTO_ICMP` (ICMP).
+  * `IPPROTO_ICMPV6` (ICMPv6).
+  * `IPPROTO_DCCP` (DCCP).
+  * `IPPROTO_SCTP` (SCTP).
+  * `IPPROTO_UDPLITE` (UDPLITE).
+  * `IPPROTO_RAW` (RAW).
+
+The userspace xtable library is used to provide help message, parse logic, sanity check, pass program specific data to the kernel space and to save the iptables state in the userspace.
+
+The arguments to the match and target callbacks have `skb` and `params`. The `skb` is the packet buffer and `params` is the program specific data passed to the kernel space from the xtables userspace library. The `skb` is a `sk_buff` structure which contains the packet data. The `params` is a `struct xt_entry_match` or `struct xt_entry_target` structure which contains the program specific data passed to the kernel space. 
+
+For the checkentry callback, the argument is a `struct xt_mtchk_param` or `struct xt_tgcheck_param` and for the destroy callback, the argument is a `struct xt_mtdtor_param` or `struct xt_tgdtor_param`.
+
+#### Example Script
+
+```lua
+local nf = require("netfilter")
+
+function drop_packet(skb, params)
+	print("drop_packet called")
+	return nf.NF_DROP
```

Comment:
probably will be simpler to use the Lunatik flags facility for your definitions.. e.g.,

```
local action = nf.action
...
function drop_packet(skb, params)
	print("drop_packet called")
	return action.DROP
```

if you think `action` is not a good name, you can call anything else..

Reply:
Looks good